engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/service/ProcessConfigEnricher.scala

```diff
@@ -0,0 +1,68 @@
+package pl.touk.nussknacker.engine.management.periodic.service
+
+import com.typesafe.config.{Config, ConfigFactory}
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.management.periodic.model.PeriodicProcessDeployment
+import pl.touk.nussknacker.engine.management.periodic.service.ProcessConfigEnricher.{DeployData, EnrichedProcessConfig, InitialScheduleData}
+import pl.touk.nussknacker.engine.marshall.ProcessMarshaller
+import pl.touk.nussknacker.engine.modelconfig.InputConfigDuringExecution
+import sttp.client.{NothingT, SttpBackend}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Use to enrich scenario config e.g. with a schedule name that can be later exposed as a global variable.
+ * Other use case could be fetching some data specific for a scenario to be used later when starting job on Flink cluster.
+ *
+ * Config enriched on initial schedule is passed to onDeploy method.
+ */
+trait ProcessConfigEnricher {
+  def onInitialSchedule(initialScheduleData: InitialScheduleData): Future[EnrichedProcessConfig]
+  def onDeploy(deployData: DeployData): Future[EnrichedProcessConfig]
+}
+
+object ProcessConfigEnricher {
+
+  trait ProcessConfigEnricherInputData {
+    def processJson: String
+    def modelConfigJson: String
+
+    def canonicalProcess: CanonicalProcess = {
+      ProcessMarshaller.fromJson(processJson).valueOr(err => throw new IllegalArgumentException(err.msg))
+    }
+
+    def modelConfig: Config = {
+      ConfigFactory.parseString(modelConfigJson)
```

Comment:
Should it be parse or load? Where/how do we resolve substitutions/defaults?

Reply:
It's rather parse. This config is sent to Flink and I think it should be only extended since factory (`ProcessConfigEnricherFactory`) has access to configuration of deployment manager. What do you think? Anyway, I think I should rename it to `InputConfigDuringExecution` here and in other classes.